### PR TITLE
[WIP] kpm fixes the order of dependencies and format in kcl.mod and kcl.mod.lock

### PR DESCRIPTION
**Pull request by officialasishkumar**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y #333  

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

**Comment by officialasishkumar** [comment]
I will do the required changes.

File: pkg/package/modfile.go [hunk 1]
```diff
@@ -146,6 +147,21 @@ type Dependencies struct {
 	Deps map[string]Dependency `json:"packages" toml:"dependencies,omitempty"`
 }
 
+// SortMapByKey will sort the dependencies by key.
+func (deps *Dependencies) SortMapByKey() {
```
[the remark this answers is not in the document]
Comment (d4v1d03) [reply pkg/package/modfile.go:151]: Hey @Peefy , i've done some changes at #334, please have a look over them if they are in the right direction.

**Review by zong-zhe — CHANGES_REQUESTED** [review]

**Comment by officialasishkumar** [comment]
@Peefy  The migration is not yet completed.

**Comment by zong-zhe** [comment]
Hi @officialasishkumar 😄

This PR conflicts with the main branch. Are you still working on it？

**Comment by officialasishkumar** [comment]
> Hi @officialasishkumar 😄
>
> This PR conflicts with the main branch. Are you still working on it？

Yeah, I will complete it. Sorry my exams were running for a few weeks. Now it's over.

**Comment by zong-zhe** [comment]
Hi @officialasishkumar 😄

I finished this issue for the 0.9.0 release, https://github.com/kcl-lang/kpm/pull/373
You'll be able to focus on the LFX now.

**Pull request closed without merging**